lib/ansible/modules/packaging/os/apt_repo.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Mikhail Gordeev
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: apt_repo
+short_description: Manage APT repositories via apt-repo
+description:
+  - Manages APT repositories using apt-repo tool.
+  - See U(https://www.altlinux.org/Apt-repo) for details about apt-repo
+notes:
+    - This module works on ALT based distros.
+version_added: "2.8"
+options:
+  repo:
+    description:
+      - Name of package to install, upgrade or remove.
+    required: true
+  state:
+    description:
+      - Indicates the desired repository state.
+    choices: [ absent, present ]
+    default: present
+  remove_others:
+    description:
+      - Remove other then added repositories
+      - Used if I(state=present)
+    type: bool
+    default: 'no'
+  update:
+    description:
+      - Update the package database after changing repositories.
+    type: bool
+    default: 'no'
+author:
+- Mikhail Gordeev (@obirvalger)
+'''
+
+EXAMPLES = '''
+- name: Remove all repositories
+  apt_repo:
+    repo: all
+    state: absent
+
+- name: Add repository `Sisysphus` and remove other repositories
+  apt_repo:
+    repo: Sisysphus
+    state: present
+    remove_others: yes
+
+- name: Add local repository `/space/ALT/Sisyphus` and update package cache
+  apt_repo:
+    repo: copy:///space/ALT/Sisyphus
+    state: present
+    update: yes
+'''
+
+RETURN = ''' # '''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+
+APT_REPO_PATH = "/usr/bin/apt-repo"
+
+
+def apt_repo(module, *args):
+    """run apt-repo with args and return its output"""
+    # make args list to use in concatenation
+    args = list(args)
+    rc, out, err = module.run_command([APT_REPO_PATH] + args)
+
+    if rc != 0:
+        module.fail_json(msg="'%s' failed: %s" % (' '.join(['apt-repo'] + args), err))
+
+    return out
+
+
+def add_repo(module, repo):
+    """add a repository"""
+    apt_repo(module, 'add', repo)
+
+
+def rm_repo(module, repo):
+    """remove a repository"""
+    apt_repo(module, 'rm', repo)
+
+
+def set_repo(module, repo):
+    """add a repository and remove other repositories"""
+    # first add to validate repository
+    apt_repo(module, 'add', repo)
+    apt_repo(module, 'rm', 'all')
+    apt_repo(module, 'add', repo)
+
+
+def update(module):
+    """update package cache"""
+    apt_repo(module, 'update')
+
+
+def main():
```

Comment:
What would be needed to support "check mode"?

Reply:
Need to add such functionality to apt-repo tool.
This work is in progress.

Reply:
@Obirvalger How would you like to progress?
Mark this PR as `WIP` (work in progress) till apt-repo tool supports this functionality
Merge this PR as-is, and update the module at a later date once apt-repo tool has been updated?

Reply:
I would like to add as-is, and update later.

Reply:
Makes sense, thank you

Reply:
You are welcome